sslcontext-kickstart-for-pem/src/main/java/nl/altindag/ssl/util/PemUtils.java

```diff
@@ -88,8 +88,15 @@
     private static final JcaX509CertificateConverter CERTIFICATE_CONVERTER = new JcaX509CertificateConverter().setProvider(BOUNCY_CASTLE_PROVIDER);
     private static final JceOpenSSLPKCS8DecryptorProviderBuilder OPEN_SSL_PKCS8_DECRYPTOR_PROVIDER_BUILDER = new JceOpenSSLPKCS8DecryptorProviderBuilder().setProvider(BOUNCY_CASTLE_PROVIDER);
     private static final JcePEMDecryptorProviderBuilder PEM_DECRYPTOR_PROVIDER_BUILDER = new JcePEMDecryptorProviderBuilder().setProvider(BOUNCY_CASTLE_PROVIDER);
-    private static final BouncyFunction<char[], InputDecryptorProvider> INPUT_DECRYPTOR_PROVIDER = password -> OPEN_SSL_PKCS8_DECRYPTOR_PROVIDER_BUILDER.build(Objects.requireNonNull(password));
-    private static final BouncyFunction<char[], PEMDecryptorProvider> PEM_DECRYPTOR_PROVIDER = password -> PEM_DECRYPTOR_PROVIDER_BUILDER.build(Objects.requireNonNull(password));
+    private static final BouncyFunction<char[], InputDecryptorProvider> INPUT_DECRYPTOR_PROVIDER = password -> OPEN_SSL_PKCS8_DECRYPTOR_PROVIDER_BUILDER.build(requirePassword(password));
+    private static final BouncyFunction<char[], PEMDecryptorProvider> PEM_DECRYPTOR_PROVIDER = password -> PEM_DECRYPTOR_PROVIDER_BUILDER.build(requirePassword(password));
+
+    private static char[] requirePassword(char[] maybeNull) {
+        if (maybeNull == null) {
+            throw new IllegalArgumentException("'password' is mandatory with an encrypted key");
```

Comment:
Ah this is much more explaining indeed. By the way I prefer to start a sentence with a capital letter. Would it be possible to adjust this to `"A password is mandatory with an encrypted key"`

Reply:
> Ah this is much more explaining indeed.

We can mimic guava's `Preconditions.checkArgument` by taking the validation message as an argument.

This would look like something like:

`ValidationUtils.requireNonNull(Object o, String errorMessage)`...

Reply:
Ah nice suggestion, but I think your current refactored solution which takes any runtime exception supplier is more flexibility. When having a method which only takes an error message we will lose this flexibility of throwing an exception defined by us.